faunadb-java-dsl/src/main/java/com/faunadb/client/types/Value.java

```diff
@@ -125,6 +125,19 @@ private Value() {
     return Decoder.decode(this, Types.hashMapOf(valueType));
   }
 
+  /**
+   * Attempts to convert the value to a {@link Map}.
+   *
+   * @param <T> the type of the values in the {@link Map}
+   * @param valueType the type of the values in the {@link Map}
+   * @return a {@link Map}, or null if the conversion failed
+   * @see Decoder
+   * @see Types
+   */
+  public final <T> Map<String, T> toMap(Class<T> valueType) {
+    return asMapOf(valueType).orNull();
```

Comment:
I throwing an exception is better than retuning a `null`. Specially because at this level, an error means a mismatch between the application code and its stored data. `null` contains very little information about that mismatch.

Reply:
I suppose I should've asked which you wanted in the ticket. Will do. :)

Reply:
It goes both ways. I could have made it more explicit. :)